ttl.py

```diff
@@ -0,0 +1,153 @@
+"""Module for TTL status management."""
+
+import asyncio
+import dataclasses
+import enum
+import time
+from typing import Any
+
+from artiq.coredevice.comm_moninj import CommMonInj, TTLOverride, TTLProbe
+
+from protocols import SortedQueue
+
+device_to_channel: dict[str, int] = {}
+channel_to_device: dict[int, str] = {}
+
+def map_device_channel(ttl_devices: list[str], device_db: dict[str, Any]):
+    """Maps TTL devices and channels.
+    
+    Args:
+        ttl_devices: See main.configs.
+        device_db: See main.device_db.
+    """
+    for device in ttl_devices:
+        channel = device_db[device]["arguments"]["channel"]
+        device_to_channel[device] = channel
+        channel_to_device[channel] = device
+
+
+@enum.unique
+class MonitorType(enum.Enum):
+    """Monitoring value type."""
+    PROBE = "probe"
+    LEVEL = "level"
+    OVERRIDE = "override"
+
+
+@dataclasses.dataclass
+class StatusType:
+    """Monitoring status type.
+    
+    Fields:
+        channel: Monitoring TTL channel number.
+        monitor_type: Monitoring value type.
+    """
+    channel: int
+    monitor_type: MonitorType
+
+    def __hash__(self) -> int:
+        """Overridden."""
+        return hash(f"{str(self.channel)}_{self.monitor_type.value}")
+
+
+# {"MonitorType value": {"TTL device name": "modified value"}}
+Modifications = dict[str, dict[str, bool]]
```

Comment:
Isn't it `dict[MonitorType, dict[str, bool]]` then?

Reply:
Umm.. A key is the value of `MonitorType`, so I set it as `str`. Isn't it correct?

Reply:
I see, then why not using `MonitorType` itself instead of its `value`?

Reply:
It is because `Modifications` is the type of data it sends to a client, so I defined it to a type which can be JSONized.